llvm/lib/Target/WebAssembly/WebAssemblyISelLowering.cpp

```diff
@@ -3239,6 +3240,55 @@ static SDValue performBitcastCombine(SDNode *N,
   return SDValue();
 }
 
+static SDValue performAnyAllCombine(SDNode *N, SelectionDAG &DAG) {
+  // any_true (setcc <X>, 0, eq) => (not (all_true X))
+  // all_true (setcc <X>, 0, eq) => (not (any_true X))
+  // any_true (setcc <X>, 0, ne) => (any_true X)
+  // all_true (setcc <X>, 0, ne) => (all_true X)
+  assert(N->getOpcode() == ISD::INTRINSIC_WO_CHAIN);
+  using namespace llvm::SDPatternMatch;
+  SDLoc DL(N);
+  static auto SimdCombiner =
+      [&](Intrinsic::WASMIntrinsics InPre, ISD::CondCode SetType,
```

Comment:
It's better to explicitly capture what you need, e.g.

```suggestion
      [&N, &DAG](Intrinsic::WASMIntrinsics InPre, ISD::CondCode SetType,
```